src/slurm_plugin/slurm_resources.py

```diff
@@ -149,48 +150,57 @@ def is_drained(self):
         drained(sinfo) is equivalent to IDLE+DRAIN(scontrol) or DOWN+DRAIN(scontrol)
         """
         return self._is_drain() and (
-            self.SLURM_SCONTROL_IDLE_STATE in self.state or self.SLURM_SCONTROL_DOWN_STATE in self.state
+            any(idel_state in self.states for idel_state in self.SLURM_SCONTROL_IDLE_STATE) or self.is_down()
         )
 
     def is_powering_down(self):
         """Check if slurm node is in powering down state."""
-        return self.SLURM_SCONTROL_POWERING_DOWN_STATE in self.state
+        return any(
+            powering_down_state in self.states for powering_down_state in self.SLURM_SCONTROL_POWERING_DOWN_STATES
+        )
 
     def is_power(self):
         """Check if slurm node is in power state."""
-        return self.SLURM_SCONTROL_POWER_STATE == self.state
+        return all(power_state in self.states for power_state in self.SLURM_SCONTROL_POWER_STATE.split("+"))
 
     def is_down(self):
         """Check if slurm node is in a down state."""
-        return self.SLURM_SCONTROL_DOWN_STATE in self.state and not self.is_powering_down()
+        return (
+            any(down_state in self.states for down_state in self.SLURM_SCONTROL_DOWN_STATES)
+            and not self.is_powering_down()
+        )
 
     def is_up(self):
         """Check if slurm node is in a healthy state."""
         return not self._is_drain() and not self.is_down() and not self.is_powering_down()
 
     def is_powering_up(self):
         """Check if slurm node is in powering up state."""
-        return self.SLURM_SCONTROL_POWER_UP_STATE in self.state
+        return any(power_up_state in self.states for power_up_state in self.SLURM_SCONTROL_POWER_UP_STATE)
```

Comment:
optional: if you want to avoid creating sets with a single element you could just do `self.SLURM_SCONTROL_POWER_UP_STATE in self.states` and remove the iteration

Reply:
Removed creating sets for all single element and changed to check `self.SLURM_SCONTROL_POWER_UP_STATE in self.states`